src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix3x2.cs

```diff
@@ -6,79 +6,67 @@
 
 namespace System.Numerics
 {
-    /// <summary>
-    /// A structure encapsulating a 3x2 matrix.
-    /// </summary>
+    /// <summary>A structure encapsulating a 3x2 matrix.</summary>
     [Intrinsic]
     public struct Matrix3x2 : IEquatable<Matrix3x2>
     {
         private const float RotationEpsilon = 0.001f * MathF.PI / 180f;     // 0.1% of a degree
 
-        #region Public Fields
-        /// <summary>
-        /// The first element of the first row
-        /// </summary>
+        private static readonly Matrix3x2 _identity = new Matrix3x2(
+            1f, 0f,
+            0f, 1f,
+            0f, 0f
+        );
+
+        /// <summary>The first element of the first row</summary>
         public float M11;
-        /// <summary>
-        /// The second element of the first row
-        /// </summary>
+
+        /// <summary>The second element of the first row</summary>
         public float M12;
-        /// <summary>
-        /// The first element of the second row
-        /// </summary>
+
+        /// <summary>The first element of the second row</summary>
         public float M21;
-        /// <summary>
-        /// The second element of the second row
-        /// </summary>
+
+        /// <summary>The second element of the second row</summary>
         public float M22;
-        /// <summary>
-        /// The first element of the third row
-        /// </summary>
+
+        /// <summary>The first element of the third row</summary>
         public float M31;
-        /// <summary>
-        /// The second element of the third row
-        /// </summary>
+
+        /// <summary>The second element of the third row</summary>
         public float M32;
-        #endregion Public Fields
 
-        private static readonly Matrix3x2 _identity = new Matrix3x2
-        (
-            1f, 0f,
-            0f, 1f,
-            0f, 0f
-        );
+        /// <summary>Constructs a Matrix3x2 from the given components.</summary>
+        public Matrix3x2(float m11, float m12,
+                         float m21, float m22,
+                         float m31, float m32)
+        {
+            M11 = m11;
+            M12 = m12;
+
+            M21 = m21;
+            M22 = m22;
+
+            M31 = m31;
+            M32 = m32;
+        }
 
-        /// <summary>
-        /// Returns the multiplicative identity matrix.
-        /// </summary>
+        /// <summary>Returns the multiplicative identity matrix.</summary>
         public static Matrix3x2 Identity
         {
-            get { return _identity; }
+            get => _identity;
         }
 
-        /// <summary>
-        /// Returns whether the matrix is the identity matrix.
-        /// </summary>
+        /// <summary>Returns whether the matrix is the identity matrix.</summary>
         public readonly bool IsIdentity
         {
-            get
-            {
-                return M11 == 1f && M22 == 1f && // Check diagonal element first for early out.
```

Comment:
Is the checking of the diagonal elements first no longer an important optimization?

Reply:
`operator ==` is itself doing a diagonal element first check.

The eventual vectorized version will just do two comparisons anyways and so at that point it won't matter.